Feat/ru_labeling

lsig opened the pull request:
# Script for getting data from label studio
In this PR request I made a script that gets data from label studio and formats it in the correct way so our model can train on the data. The script is type annotated and required me to add a dotenv dependency for loading the the api key from the .env file. 

# Testing 
- Ran the script on local machine with my api_key which produced the desired output 
- Ran format_and_lint before committing 
- No unit tests

# Bugs 
- Mypy complains: Incompatible types (expression has type "MetaInfo", TypedDict item "meta" has type "MetaInfo") in line 28 even though the meta item only has the type MetaInfo, but nonetheless the script runs as expected.

lsig committed:
Script and type declarations for data from labeling studion made, works
calling the api with a correct key

lsig committed:
Added dotenv dependency

lsig committed:
Removed testing call

lsig committed:
Changed after Codacy complained

lsig committed:
Forgot to run format_and_lint

Eysteinn-Orn commented on .gitignore:
```diff
@@ -0,0 +1 @@
+__pycache__/
```
We probably should add this to the repository, possibly could be its own PR. But this is also good!

Eysteinn-Orn commented on training_datasets/ruquad_labeling.py:
```diff
@@ -0,0 +1,122 @@
+    return all_project_records
+
+
+def __invalid_record(record: CleanRUResponse) -> bool:
```
A better name for this would be __is_valid_record to indicate that this is a boolean function more-so than just the type hint

Eysteinn-Orn replied:
I think you can refactor this function to be more readable;
Maybe something like this:
```python
def __is_invalid_record(record: CleanRUResponse) -> bool:
    annotations = record["annotations"]
    if not annotations or not annotations[0]["result"]:
        return True

    result = annotations[0]["result"][0]["value"]
    if "labels" not in result or result["labels"][0] == "Archive":
        return True

    paragraph = record["meta"]["paragraph"]
    if len(paragraph.split(" ")) > 300:
        return True

    return False
```

lsig replied:
I feel like the context of the function (where it is used) makes it obvious. if it is an invalid record, then we skip it.

Eysteinn-Orn commented on training_datasets/ruquad_labeling.py:
```diff
@@ -0,0 +1,122 @@
+    return api_key
+
+
+def __records(project_id: int, api_key: str) -> list[CleanRUResponse]:
```
This could also be renamed to __get_records or __get_clean_records. This is optional.

Eysteinn-Orn reviewed (approved):
Overall, looks good and a good contribution!

lsig committed:
Changed the structure of the data to return a dict sorted by sources
making it easier to work with the powersets of the data, if this is not
satisfiable it can always be reverted to the last commit or changed

lsig committed:
Restructured data after recomondations from team, much more efficent
grouping of related data. Code also made more efficient by change.

njallskarp reviewed (approved):
very nice. I like
